robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,222 @@
+import logging
+import re
+
+import requests
+from tenacity import retry
+from tenacity import stop_after_attempt
+from tenacity import wait_fixed
+
+from robottelo.config import settings
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ReportPortal:
+    """Represents ReportPortal
+
+    This holds the properties and functions to interact with Report Portal properties and launches
+    """
+
+    def __init__(self):
+        """Configure the settings and initiate report portal properties"""
+        if not settings.configured:
+            settings.configure()
+        self.rp_url = settings.report_portal.rp_url
+        self.rp_project = settings.report_portal.rp_project
+        self.rp_api_key = settings.report_portal.rp_key
+
+    @property
+    def api_url(self):
+        """Super url of report portal
+        :returns: Base url for API request
+        """
+        return f'{self.rp_url}/api/v1/{self.rp_project}/'
+
+    @property
+    def headers(self):
+        """The headers for Report Portal Requests.
+        :returns: header for API request
+        """
+        return {'Authorization': f'Bearer {self.rp_api_key}'}
+
+    def launches(self, sat_version=None, launch_type='satellite6', latest=200):
+        """Returns launches in Report Portal customized by sat_version, launch_type and
+        latest number of launches sorted by latest sat version/snap version.
+
+        This does not includes each tests data for all tests in all launches, but it includes
+        just an overview data for all and each launch in Report Portal
+
+        :param str sat_version: The satellite version
+        :param str launch_type: Either satellite6 or upgrades,
+            default returns only non-upgrade launches
+        :param int latest: The latest number of launches dict to return,
+            default returns latest 200 snaps else the max available
+        :returns dict: The directory of launches keyed by satellite version
+        """
+        launch_types = ['satellite6', 'upgrades']
+        if launch_type not in launch_types:
+            raise ValueError(
+                f'Wrong Launch Type is given as {launch_type} should be one of {launch_types}'
+            )
+        params = {'page.page': 1, 'page.size': 500, 'page.sort': 'start_time'}
+        resp = requests.get(
+            url=f'{self.api_url}/launch', headers=self.headers, params=params, verify=False
+        )
+        resp.raise_for_status()
+        data = resp.json()
+        launches_ = {}
+        if data['content']:
+            all_launches = sorted(
+                data['content'], key=lambda launch: launch['start_time'], reverse=True
+            )
+            typed_launches = list(
```

Comment:
I think this is being cast directly to a list because of slicing it in the `for` statement. The iterator could be used directly if we aren't slicing, as mentioned above.

Reply:
Not sure about the alternative to slice the filter object. Do you think of any?